packages/@o3r/rules-engine/src/engine/helpers/filter-ruleset-event.operator.spec.ts

```diff
@@ -134,68 +147,66 @@ describe('Filter rulesets event operator', () => {
   }, {});
 
   const firstValue = rulesets.reduce<Record<string, RulesetExecutor>>((accRuleset, ruleset) => {
-    accRuleset[ruleset.id] = new RulesetExecutor(ruleset, {retrieveOrCreateFactStream: () => of(undefined), operators} as any);
+    accRuleset[ruleset.id] = new RulesetExecutor(ruleset, { retrieveOrCreateFactStream: () => of(undefined), operators } as any);
     return accRuleset;
   }, {});
 
   const rulesetsMapSubject$ = new BehaviorSubject<Record<string, RulesetExecutor>>(firstValue);
 
-  test('should consider only first ruleset', (done) => {
-
-    rulesetsMapSubject$.pipe(
-      filterRulesetsEventStream(['ruleset1'])
-    ).subscribe(data => {
-      expect(data.length).toBe(2);
-      done();
+  test('should consider only first ruleset', () => {
+    return new Promise((done) => {
+      rulesetsMapSubject$.pipe(
+        filterRulesetsEventStream(['ruleset1'])
+      ).subscribe((data) => {
+        expect(data.length).toBe(2);
+        done();
+      });
     });
-
   });
 
-  test('should consider only second ruleset', (done) => {
-
-    rulesetsMapSubject$.pipe(
-      filterRulesetsEventStream(['ruleset2'])
-    ).subscribe(data => {
-      expect(data.length).toBe(1);
-      done();
+  test('should consider only second ruleset', () => {
+    return new Promise((done) => {
+      rulesetsMapSubject$.pipe(
+        filterRulesetsEventStream(['ruleset2'])
+      ).subscribe((data) => {
+        expect(data.length).toBe(1);
+        done();
+      });
     });
-
   });
 
-  test('should consider all rulesets by not passing any filter', (done) => {
-
-    rulesetsMapSubject$.pipe(
-      filterRulesetsEventStream()
-    ).subscribe(data => {
-      expect(data.length).toBe(3);
-      done();
+  test('should consider all rulesets by not passing any filter', () => {
+    return new Promise((done) => {
+      rulesetsMapSubject$.pipe(
+        filterRulesetsEventStream()
+      ).subscribe((data) => {
+        expect(data.length).toBe(3);
+        done();
+      });
     });
-
   });
 
-  test('should consider all rulesets ids passed', (done) => {
-
-    rulesetsMapSubject$.pipe(
-      filterRulesetsEventStream(['ruleset1', 'ruleset2'])
-    ).subscribe(data => {
-      expect(data.length).toBe(3);
-      done();
+  test('should consider all rulesets ids passed', () => {
+    return new Promise((done) => {
+      rulesetsMapSubject$.pipe(
+        filterRulesetsEventStream(['ruleset1', 'ruleset2'])
+      ).subscribe((data) => {
+        expect(data.length).toBe(3);
+        done();
+      });
     });
-
   });
 
   test('should not emit if ruleset id does not match any registered ruleset', async () => {
-
     let emittedActions: ActionBlock[] | undefined;
 
     rulesetsMapSubject$.pipe(
       filterRulesetsEventStream(['ruleset3'])
-    ).subscribe(data => {
+    ).subscribe((data) => {
       emittedActions = data;
     });
 
     await jest.advanceTimersByTimeAsync(500);
     expect(emittedActions).toBe(undefined);
```

Comment:
Not really relative to the fix but this can lead to issue because there is no unsubscription.
Maybe it can be fixed with:

```typescript
     const stop = new Subject<void>();
     const data = firstValueFrom(
       rulesetsMapSubject$.pipe(
         takeUntil(stop),
         filterRulesetsEventStream(['ruleset3'])
       ), {defaultValue: 'no emit'}
    );

    await jest.advanceTimersByTimeAsync(500);
    stop.next();
    await expect(data).resolves.toBe('no emit');
```

Reply:
`firstValueFrom` should be enough to close the subscription, I don't think you need the `takeUntil`

Reply:
It will lead to a promise never resolve, I don't know if the subscription is close when there is no ref to this promise anymore.
Maybe it is also simpler to check the `non-fulfilled` status by explicitly completing the observable

Reply:
my bad, I didn't realize that we were checking that the observable doesn't emit